bindings/go/mio/mio.go

```diff
@@ -369,11 +369,19 @@ func (mio *Mio) getOptimalBlockSz(bufSz int) (bsz, gsz int) {
                   pa.pa_P, pa.pa_N, pa.pa_K, pa.pa_S,
                            pa.pa_N + pa.pa_K + pa.pa_S)
     }
+
     usz := int(C.m0_obj_layout_id_to_unit_size(mio.objLid))
     gsz = usz * int(pa.pa_N) // group size in data units only
-    // should be max 2-times pool-width deep, otherwise we may get -E2BIG
-    maxBs := int(C.uint(usz) * 2 * pa.pa_P * pa.pa_N /
-                    (pa.pa_N + pa.pa_K + pa.pa_S))
+
+    // bs should be max 4-times pool-width deep counting by 1MB units or
+    // 8-times deep counting by 512K units, 16-times deep by 256K units,
+    // and so on. Several units to one target will be aggregated to make
+    // fewer network RPCs, disk i/o operations and BE transactions.
+    k := C.uint(128 / ((usz + 0x7fff) / 0x8000))
```

Comment:
A question: for unit size less than 32KB, the k is the same as 32KB, can you explain the reason behind this?

Reply:
You probably mean why k is 128 in this case? Well, it's the maximum number of packing the units into one target: 128 * 32K == 4MB. And we don't want bigger sizes to go per target: as practice with LNet shows - it only slows down the performance. (LNet frame is only 1MB.)

Hope it helps.

Reply:
Added:
```
+    // For unit sizes of 32K or less, the koefficient (k) is 128, which
+    // makes it 32K * 128 == 4MB - the maximum amount per target when
+    // the performance is still good.
```